### PR TITLE
QF-1690 Track engagement links click

### PR DESCRIPTION
**Pull request by ChrisJWoodcock**

### CODE REVIEW

File: src/SFA.DAS.ApprenticeFeedback.Api/Controllers/FeedbackTransactionController.cs [hunk 1]
```diff
@@ -73,5 +73,23 @@ public async Task<IActionResult> ProcessEmailTransaction(int feedbackTransaction
...
 
         }
+
+        [HttpPost("{feedbackTransactionId}/track-click")]
+        public async Task<IActionResult> TrackEmailTransactionClick(long feedbackTransactionId, [FromBody] TrackEmailTransactionClickCommand command)
+        {
+            try
+            {
+                var result = await _mediator.Send(command);
+                if (null == result) return NotFound();
```
Comment (SawNawDfE) [inline src/SFA.DAS.ApprenticeFeedback.Api/Controllers/FeedbackTransactionController.cs:83]: The handler for this command doesn't seem to return null?

File: src/SFA.DAS.ApprenticeFeedback.Application.UnitTests/Commands/WhenHandlingTrackEmailTransactionClickCommand.cs [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+            command.LinkName = "SomeLinkName";
+            command.LinkUrl = "https://somelink.com";
+            command.ClickedOn = DateTime.Now;
+
+            // Act
+            await handler.Handle(command, CancellationToken.None);
+
+            // Assert
+            context.FeedbackTransactionClicks.Should().HaveCount(1);
+            context.FeedbackTransactionClicks.First().Should().BeEquivalentTo(new FeedbackTransactionClick
```
Comment (SawNawDfE) [inline src/SFA.DAS.ApprenticeFeedback.Application.UnitTests/Commands/WhenHandlingTrackEmailTransactionClickCommand.cs:139]: If the `First()` on line 139 becomes a `Single()`, then line 138 can be deleted

**Review by SawNawDfE — COMMENTED** [review]
Left a few comments

File: src/SFA.DAS.ApprenticeFeedback.Api/Controllers/FeedbackTransactionController.cs (hunk 1 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.ApprenticeFeedback.Api/Controllers/FeedbackTransactionController.cs]: Removed

File: src/SFA.DAS.ApprenticeFeedback.Application.UnitTests/Commands/WhenHandlingTrackEmailTransactionClickCommand.cs (hunk 2 above)
Comment (ChrisJWoodcock) [reply src/SFA.DAS.ApprenticeFeedback.Application.UnitTests/Commands/WhenHandlingTrackEmailTransactionClickCommand.cs]: Changed to Single

**Review by SawNawDfE — APPROVED** [review]

**Review by Jordan-fujitsu — APPROVED** [review]

**Pull request merged**